docs/data-source-matrix.md

```diff
@@ -0,0 +1,54 @@
+# Data Source Matrices
+
+The Carbon Aware SDK includes access to various data sources of carbon aware data, including WattTime, ElectricityMaps, and a custom JSON source. These matrices are an attempt to track what features of the Carbon Aware SDK are enabled for which data sources.
+
+## Contents
+
+- [Type of Data Sources and Configuration](#type-of-data-sources-and-configuration)
+- [Data Source Routes Available](#data-source-routes-available)
+- [Location Source Availability](#location-source-availability)
+
+## Type of Data Sources and Configuration
+
+In the CarbonAware SDK configuration, you can set what data source to use as the `EmissionsDataSource` and the `ForecastDataSource`. There are also certain configuration fields that must be set in order to access the raw data.
+| Type      | WattTime  | ElectricityMaps | JSON |
+|--------------------------|-----------|-----------------|------|
+| Is Emissions DataSource   | Yes      | Yes             | Yes     |
+| Is Forecast DataSource   | Yes  | Yes            | No     |
+| Makes HTTP(s) call   | Yes  | Yes            | No    |
+| Can Use Custom Data   | No  | No            | Yes    |
+| Needs Authentication Config   | Yes - username/password | Yes - token/header            | No    |
```

Comment:
For the purpose of this comparison, does the user need to be aware of the type of auth?  If no, consider removing.  If yes, consider referencing the nuance as a footnote.

Same comment applies to this entire matrix